datalab/mlalpha/commands/_ml.py

```diff
@@ -0,0 +1,160 @@
+  preprocess_parser.add_argument('--cloud',
+                                 help='Whether to run the preprocessing job in the cloud.',
+                                 action='store_true', default=False)
+  preprocess_parser.add_argument('--package',
+                                 help='The preprocess package to use. Can be a gs or local path.',
+                                 required=True)
+  preprocess_parser.set_defaults(func=_preprocess)
+
+  train_parser = parser.subcommand('train', 'Train an ML model.')
+  train_parser.add_argument('--usage',
+                            help='Show usage from the specified trainer package',
+                            action='store_true', default=False)
+  train_parser.add_argument('--cloud',
+                            help='Whether to run the training job in the cloud.',
+                            action='store_true', default=False)
+  train_parser.add_argument('--package',
+                            help='The trainer package to use. Can be a gs or local path.',
+                            required=True)
+  train_parser.set_defaults(func=_train)
+
+  predict_parser = parser.subcommand('predict', 'Predict with an ML model.')
+  predict_parser.add_argument('--usage',
+                              help='Show usage from the specified prediction package',
+                              action='store_true', default=False)
+  predict_parser.add_argument('--cloud',
+                              help='Whether to run prediction in the cloud.',
+                              action='store_true', default=False)
+  predict_parser.add_argument('--package',
+                              help='The prediction package to use. Can be a gs or local path.',
+                              required=True)
+  predict_parser.set_defaults(func=_predict)
+
+  batch_predict_parser = parser.subcommand('batch_predict', 'Batch predict with an ML model.')
+  batch_predict_parser.add_argument('--usage',
+                                    help='Show usage from the specified prediction package',
+                                    action='store_true', default=False)
+  batch_predict_parser.add_argument('--cloud',
+                                    help='Whether to run prediction in the cloud.',
+                                    action='store_true', default=False)
+  batch_predict_parser.add_argument('--package',
+                                    help='The prediction package to use. Can be a gs or local path.',
+                                    required=True)
+  batch_predict_parser.set_defaults(func=_batch_predict)
+
+  namespace = datalab.utils.commands.notebook_environment()
+  return datalab.utils.commands.handle_magic_line(line, cell, parser, namespace=namespace)
+
+
+def _command_template(package_uri, func_name):
+  """Return (args_list, docstring).
+     args_list is in the form of:
+       arg1:
+       arg2:
+       arg3: (optional)
+  """
+  pr = datalab.mlalpha.PackageRunner(package_uri)
```

Comment:
The "finally" blocks in the functions should take care of it. But now I am using it in with statement so it should be good too.